ui/components/app/currency-input/currency-input.js

```diff
@@ -41,84 +45,93 @@ export default function CurrencyInput({
   onPreferenceToggle,
   swapIcon,
   className = '',
+  // if null, the asset is the native currency
+  asset,
 }) {
   const t = useContext(I18nContext);
 
+  const assetDecimals = asset?.decimals || NATIVE_CURRENCY_DECIMALS;
+
   const preferredCurrency = useSelector(getNativeCurrency);
   const secondaryCurrency = useSelector(getCurrentCurrency);
-  const conversionRate = useSelector(getConversionRate);
+
+  const primarySuffix =
+    asset?.symbol || preferredCurrency || EtherDenomination.ETH;
+  const secondarySuffix = secondaryCurrency.toUpperCase();
+  const isLongSymbol = (primarySuffix?.length || 0) > LARGE_SYMBOL_LENGTH;
+
+  const [shouldDisplayFiat, setShouldDisplayFiat] = useState(featureSecondary);
   const showFiat = useSelector(getShouldShowFiat);
+  const hideSecondary = !showFiat;
+  const shouldUseFiat = hideSecondary ? false : Boolean(shouldDisplayFiat);
+  const isTokenPrimary = !shouldUseFiat;
```

Comment:
Nit: while you're refactoring this component, I think the naming here could be further improved. At first glance it's hard to understand the difference between `shouldDisplayFiat`, `showFiat` and `shouldUseFiat`. Comments or making the names more descriptive would be helpful

Reply:
Great callout – refactored

Reply:
`shouldDisplayFiatIfAvailable` doesn't check for "availability" of fiat since it doesn't take conversion rate into account, just the user's settings/preferences. `isFiatPreferred` would be more accurate